File: src/lib/money.ts
const CENTS_PER_UNIT = 100;

const yuanFormatter = new Intl.NumberFormat('zh-CN', {
  style: 'decimal',
  minimumFractionDigits: 2,
  maximumFractionDigits: 2,
});

const compactFormatter = new Intl.NumberFormat('zh-CN', {
  notation: 'compact',
  maximumFractionDigits: 1,
});

export type FormatMoneyOptions = {
  showSymbol?: boolean;
  signed?: boolean;
  compact?: boolean;
};

export function formatMoney(cents: number, options: FormatMoneyOptions = {}): string {
  const { showSymbol = true, signed = false, compact = false } = options;

  const yuan = cents / CENTS_PER_UNIT;
  const absYuan = Math.abs(yuan);

  const body = compact ? compactFormatter.format(absYuan) : yuanFormatter.format(absYuan);

  const sign = signed ? (yuan >= 0 ? '+' : '-') : yuan < 0 ? '-' : '';
  const symbol = showSymbol ? '¥' : '';

  return `${sign}${symbol}${body}`;
}

export function parseMoneyToCents(input: string): number | null {
  const trimmed = input.replace(/[¥,\s元]/g, '').trim();
  if (trimmed.length === 0) return null;
  const num = Number(trimmed);
  if (!Number.isFinite(num)) return null;
  return Math.round(num * CENTS_PER_UNIT);
}

export function centsToYuan(cents: number): number {
  return cents / CENTS_PER_UNIT;
}

export function yuanToCents(yuan: number): number {
  return Math.round(yuan * CENTS_PER_UNIT);
}

File: CashFlows/Lib/Money.swift
import Foundation

enum Money {
    private static let yuanFormatter: NumberFormatter = {
        let f = NumberFormatter()
        f.numberStyle = .decimal
        f.minimumFractionDigits = 2
        f.maximumFractionDigits = 2
        f.locale = Locale(identifier: "zh_CN")
        return f
    }()

    static func format(cents: Int, showSymbol: Bool = true, signed: Bool = false) -> String {
        let yuan = Decimal(cents) / 100
        let absValue = yuan.magnitude
        let body = yuanFormatter.string(from: absValue as NSDecimalNumber) ?? "0.00"
        let prefix: String = {
            if signed { return cents >= 0 ? "+" : "-" }
            return cents < 0 ? "-" : ""
        }()
        let symbol = showSymbol ? "¥" : ""
        return "\(prefix)\(symbol)\(body)"
    }

    static func parse(_ input: String) -> Int? {
        let trimmed = input
            .replacingOccurrences(of: "¥", with: "")
            .replacingOccurrences(of: ",", with: "")
            .replacingOccurrences(of: "元", with: "")
            .trimmingCharacters(in: .whitespaces)
        guard !trimmed.isEmpty,
              let value = Decimal(string: trimmed)
        else { return nil }
        let cents = (value * 100) as NSDecimalNumber
        return cents.intValue
    }
}
